symbulate/results.py

```diff
@@ -333,56 +358,106 @@ def plot(self, type=None, alpha=None, normalize=True, jitter=False,
                     y += np.random.normal(loc=0, scale=.01 * (max(y) - min(y)), size=len(y))
-                # get next color in cycle
-                axes = plt.gca()
-                color = get_next_color(axes)
-                plt.scatter(x, y, color=color, alpha=alpha, **kwargs)
-            elif type == "tile" and discrete_x and discrete_y:
+                nullfmt = NullFormatter() #removes labels on fig
+                fig, ax = plt.subplots(1, 1)
+                ax.scatter(x, y, alpha=alpha, c='b')
+            elif all(x in ("scatter", "marginal") for x in type):
+                if jitter:
+                    x += np.random.normal(loc=0, scale=.01 * (max(x) - min(x)), size=len(x))
+                    y += np.random.normal(loc=0, scale=.01 * (max(y) - min(y)), size=len(y))
+                nullfmt = NullFormatter() #removes labels on fig
+                fig = plt.figure()
+                gs = GridSpec(4, 4)
+                ax_joint = fig.add_subplot(gs[1:4, 0:3])
+                ax_marg_x = fig.add_subplot(gs[0, 0:3])
+                ax_marg_y = fig.add_subplot(gs[1:4, 3])
+                ax_joint.scatter(x, y, alpha=alpha)
+                ax_marg_x.hist(x)
+                ax_marg_y.hist(y, orientation='horizontal')
+                plt.setp(ax_marg_x.get_xticklabels(), visible=False)
+                plt.setp(ax_marg_y.get_yticklabels(), visible=False)
+            elif all(x in ("hist2d", ) for x in type) and continuous_both:
+                nullfmt = NullFormatter() #removes labels on fig
+                fig, ax = plt.subplots(1, 1)
+                ax.hist2d(x, y, bins=bins, cmap='Blues')
+            elif all(x in ("hist2d", "marginal") for x in type) and continuous_both:
+                nullfmt = NullFormatter() #removes labels on fig
+                fig = plt.figure()
+                gs = GridSpec(4, 4)
+                ax_joint = fig.add_subplot(gs[1:4, 0:3])
+                ax_marg_x = fig.add_subplot(gs[0, 0:3])
+                ax_marg_y = fig.add_subplot(gs[1:4, 3])
+                ax_joint.hist2d(x, y, bins=bins, cmap='Blues')
+                ax_marg_x.hist(x, color='b')
+                ax_marg_y.hist(y, color='b', orientation='horizontal')
+                plt.setp(ax_marg_x.get_xticklabels(), visible=False)
+                plt.setp(ax_marg_y.get_yticklabels(), visible=False)
+            elif all(x in ("tile", ) for x in type) and discrete_both:
                 res = pd.DataFrame({'X': x, 'Y': y})
                 res['num'] = 1
-                temp = pd.pivot_table(res, values = 'num', index = ['Y'],
-                    columns = ['X'], aggfunc = np.sum)
-                temp = temp / len(x)
-                sns.set()
-                sns.cubehelix_palette(8)
+                res_pivot = pd.pivot_table(res, values='num', index=['Y'],
+                    columns=['X'], aggfunc=np.sum)
+                res_pivot = res_pivot / len(x)
+                res_pivot[np.isnan(res_pivot)] = 0
```

Comment:
If this is a Pandas DataFrame, you should do `.fillna(0)`. But as discussed elsewhere, you may want to eliminate the use of Pandas altogether.